apps/system/utils/kdbg_stackopt.c

```diff
@@ -0,0 +1,194 @@
+
+/****************************************************************************
+ * Extern Data
+ ****************************************************************************/
+extern struct stkmon_save_s stkmon_arr[CONFIG_MAX_TASKS * 2];
+extern tash_taskinfo_t tash_taskinfo_list[];
+
+bool find_app(TASH_CMD_CALLBACK *cb, char *name)
+{
+	int task_idx;
+	bool flag = false;
+	char str[CONFIG_TASK_NAME_SIZE + 1];
+	int count = tash_get_cmdscount();
+	
+	for (task_idx = 0; task_idx < count; task_idx++) {
+		tash_get_cmdpair(str, cb, task_idx);
+		if (!strncmp(name, str, CONFIG_TASK_NAME_SIZE)) {
+			flag = true;
+			break;
+		}
+	}
+	return flag;
+}
+
+int execute_app(int argc, char **args)
+{
+	int app_idx;
+	TASH_CMD_CALLBACK cb;
+	int pid = 0;
+	int pri = CONFIG_TASH_CMDTASK_PRIORITY;
+	long stack_size = CONFIG_TASH_CMDTASK_STACKSIZE;
+
+	if (find_app(&cb, args[0])) {
+#if defined(CONFIG_BUILTIN_APPS)
+		for (app_idx = 0; tash_taskinfo_list[app_idx].str != NULL; app_idx++) {
+			if (!strncmp(args[0], tash_taskinfo_list[app_idx].str, CONFIG_TASK_NAME_SIZE)) {
+				pri = tash_taskinfo_list[app_idx].task_prio;
+				stack_size = tash_taskinfo_list[app_idx].task_stacksize;
+				break;
+			}
+		}
+#endif
+		pid = task_create(args[0], pri, stack_size, cb , &args[1]);
+	}
+	return pid;
+}
+
+int get_frame_size(int argc, char **args)
+{
+	int args_idx = 0;
+	size_t strtablen = 0;
+	size_t argvlen = (argc + 1) * sizeof(FAR char *);
+	size_t frame_size = 0;
+	
+	while (args[args_idx]) {
+		strtablen += (strlen(args[args_idx]) + 1);
+		args_idx++;
+	}
+	
+	frame_size = STACK_ALIGN_UP(argvlen + strtablen);
+	return frame_size;
+}
+
+void print_measure_result(pid_t pid, int argc, char **args)
+{
+	int stkmon_idx;
+	for (stkmon_idx = 0; stkmon_idx < CONFIG_MAX_TASKS * 2; stkmon_idx++) {
+		if (stkmon_arr[stkmon_idx].chk_pid == pid) {
+			break;
+		}
+	}
+
+	printf("-------------------------------------------------------\n");
+	printf(">> Stack Size Measure Finish\n");
+	printf("APP Name : %s\n", stkmon_arr[stkmon_idx].chk_name);
+	printf("Allocate %d stack size for measure\n",  stkmon_arr[stkmon_idx].chk_stksize);
+	printf("Optimal Stack Size is %d\n", stkmon_arr[stkmon_idx].chk_peaksize + get_frame_size(argc, args));
+	printf("If the arguments change, optimal stack size will change\n");
+	printf("-------------------------------------------------------\n");
+	return;
+}
+
+int kdbg_stackopt(int argc, char **args)
+{
+	int pid;
+	int status;
+
+	if (argc <= 1) {
+		printf("Invalid argument\n");
+		goto usage;
+	}
+	
+	if (!strcmp(args[1], "--help")) {
+		goto usage;
+	}
+	
+	if ((pid = execute_app(argc - 1, &args[1])) < 0) {
+		printf("Cannot create task\n");
+		goto usage;
+	} else if (pid == 0) {
+		printf("Cannot find APP\n");
+		goto usage;
+	} else {
+		waitpid(pid, &status, 0);
+		print_measure_result((pid_t)pid, argc - 1, &args[1]);
+	}
+	return OK;
+
+usage:
+	printf("\nUsage: stkprof NAME [ARGS]\n");
```

Comment:
change "stkprof" to "stkopt"
do you have any special reason that use [ ] only for ARGS?